$NetBSD: patch-meson.build,v 1.5 2025/07/19 13:01:10 gutteridge Exp $

Remove asserts that break portability.

Fix Solaris derivate builds, where defining _XOPEN_SOURCE_EXTENDED can be
harmful, avoid _XOPEN_SOURCE clashes, etc.

--- meson.build.orig	2023-06-01 19:33:31.000000000 +0000
+++ meson.build
@@ -222,7 +222,7 @@ libc_feature_defines = [
 
 system = host_machine.system()
 
-if system == 'freebsd'
+if system == 'freebsd' or system == 'sunos'
   # Defining _POSIX_C_SOURCE above makes freebsd not expose some functionality
   # that's hidden behind __BSD_VISIBLE.  Not defininy any of the above however
   # makes it expose verything.
@@ -603,14 +596,10 @@ linker_flags = [
 foreach flag: linker_flags
   if cc.has_link_argument(flag[0])
     add_project_link_arguments(flag[0], language: 'c')
-  elif flag[1]
-    assert(false, flag[0] + ' is required but not supported')
   endif
 
   if cxx.has_link_argument(flag[0])
     add_project_link_arguments(flag[0], language: 'cpp')
-  elif flag[1]
-    assert(false, flag[0] + ' is required but not supported')
   endif
 endforeach
 
